praw/exceptions.py

```diff
@@ -7,32 +7,85 @@
 All other exceptions are subclassed from :class:`.ClientException`.
 
 """
-from typing import Optional
+from typing import List, Optional, TypeVar, Union
+
+_RedditErrorItem = TypeVar("_RedditErrorItem")
 
 
 class PRAWException(Exception):
     """The base PRAW Exception that all other exception classes extend."""
 
 
-class APIException(PRAWException):
-    """Indicate exception that involve responses from Reddit's API."""
+class RedditErrorItem:
+    """Represents a single error returned from Reddit's API."""
 
-    def __init__(self, error_type: str, message: str, field: Optional[str]):
-        """Initialize an instance of APIException.
+    @property
+    def error_message(self):
+        """Get the completed error message string."""
+        error_str = "{}: '{}'".format(self.error_type, self.message)
+        if self.field:
+            error_str += " on field '{}'".format(self.field)
+        return error_str
+
+    def __init__(
+        self, error_type: str, message: str, field: Optional[str] = None
+    ):
+        """Instantize an error item.
 
         :param error_type: The error type set on Reddit's end.
         :param message: The associated message for the error.
         :param field: The input field associated with the error if available.
         """
-        error_str = "{}: '{}'".format(error_type, message)
-        if field:
-            error_str += " on field '{}'".format(field)
-
-        super().__init__(error_str)
         self.error_type = error_type
         self.message = message
         self.field = field
 
+    def __eq__(self, other: Union[_RedditErrorItem, List[str]]):
+        """Check for equality."""
+        if isinstance(other, RedditErrorItem):
+            return (self.error_type, self.message, self.field) == (
+                other.error_type,
+                other.message,
+                other.field,
+            )
+        elif isinstance(other, list):
+            return [self.error_type, self.message, self.field] == other
+        return False
```

Comment:
This should return False, because any equality comparison to any other type (int, float, etc.) should be False.

Reply:
Unless the other object says otherwise. It may not even return a boolean. Returning `NotImplemented` gives the other object an opportunity to compare if it knows how.

https://stackoverflow.com/a/879005

> `NotImplemented` signals to the runtime that it should ask someone else to satisfy the operation. In the expression `a == b`, if `a.__eq__(b)` returns `NotImplemented`, then Python tries `b.__eq__(a)`. If `b` knows enough to return `True` or `False`, then the expression can succeed. If it doesn't, then the runtime will fall back to the built-in behavior (which is based on identity for `==` and `!=`).